components-api/src/main/scala/pl/touk/nussknacker/engine/api/Context.scala

```diff
@@ -27,18 +27,23 @@ object Context {
   def apply(id: String, variables: Map[String, Any]): Context =
     Context(id, variables, None)
 
+  def apply(id: String, variables: Map[String, Any], parentContext: Option[Context]): Context =
+    Context(id, id, variables, parentContext)
+
 }
 
 case class ContextId(value: String)
 
 /**
  * Context is container for variables used in expression evaluation
  *
+ * @param initialId     the initial id of the context (id may change during the processing, but the initial id is preserved in this field
  * @param id            correlation id/trace id used for tracing (logs, error presentation) and for tests mechanism, it should be always defined
  * @param variables     variables available in evaluation
  * @param parentContext context used for scopes handling, mainly for fragment invocation purpose
  */
 case class Context(
+    initialId: String,
```

Comment:
This change will no longer be needed after the ContextId refactor. The initialId is basically ContextId without prefix.